velox/connectors/hive/HiveDataSink.cpp

```diff
@@ -70,33 +123,86 @@ std::unique_ptr<velox::dwrf::Writer> HiveDataSink::createWriter() {
   // Without explicitly setting flush policy, the default memory based flush
   // policy is used.
 
-  auto hiveWriterParameters =
-      std::dynamic_pointer_cast<const HiveWriterParameters>(
-          writeProtocol_->getWriterParameters(
-              insertTableHandle_, connectorQueryCtx_));
-  VELOX_CHECK_NOT_NULL(
-      hiveWriterParameters,
-      "Hive data sink expects write parameters for Hive.");
-  writerParameters_.emplace_back(hiveWriterParameters);
-
-  auto writePath = fs::path(hiveWriterParameters->writeDirectory()) /
-      hiveWriterParameters->writeFileName();
+  auto writePath = fs::path(writerParameters->writeDirectory()) /
+      writerParameters->writeFileName();
   auto sink = dwio::common::DataSink::create(writePath);
   return std::make_unique<Writer>(
       options, std::move(sink), *connectorQueryCtx_->memoryPool());
 }
 
+std::shared_ptr<const HiveWriterParameters> HiveDataSink::getWriterParameters(
+    const std::optional<std::string>& partition) const {
+  auto hiveTableWriteHandle =
+      std::dynamic_pointer_cast<const HiveInsertTableHandle>(
+          insertTableHandle_);
+  VELOX_CHECK_NOT_NULL(
+      hiveTableWriteHandle, "Hive connector expects Hive insert table handle.");
+
+  std::string targetFileName;
+  std::string writeFileName;
+  switch (commitStrategy_) {
+    case CommitStrategy::kNoCommit: {
+      targetFileName = fmt::format(
+          "{}_{}_{}",
+          connectorQueryCtx_->taskId(),
+          connectorQueryCtx_->driverId(),
+          makeUuid());
+      writeFileName = targetFileName;
+      break;
+    }
+    case CommitStrategy::kTaskCommit: {
+      targetFileName = fmt::format(
```

Comment:
I'd like to get a better understanding of this naming scheme. In particular, it seems to me that taskId is globally unique, hence, taskId + driverId + 0 is unique. If that's the case, why is it necessary to add uuid when making writeFileName? Do we expect some garbage collection logic to look at files named .tmp.velox.xxx?

Reply:
Hi @mbasmanova , the uuid part is to handle the case of task retry and speculative execution, so that the file names won't conflict during task retry. What I know about is that coordinator side (probably through warm storage service) will ignore files prefixed with dot, as been told many times.